orttraining/orttraining/test/graph/gradient_graph_builder_test.cc

```diff
@@ -1127,31 +1127,49 @@ TEST(GradientGraphBuilderTest, PipelineOnlinePartition) {
   pipe.cut_list.emplace_back(cut0);
   pipe.cut_list.emplace_back(cut1);
 
-  for (int i = 0; i < 3; ++i) {
+  TrainingSession::TrainingConfiguration::MixedPrecisionConfiguration mixed_precision_config{};
+  mixed_precision_config.use_fp16_initializers = true;
+
+  // 2 test variations - full precision (m == 0) and mixed precision (m == 1)
+  for (int m = 0; m < 2; ++m) {
```

Comment:
I'd suggest not to use counter to control which config to use. Consider to use vector of configs, and then use the counter to get the config setting. say, configs[m]. Or a wrapper call from another function to call this test using flag, ex, true/false
TEST(TrainableDropoutTest, RatioLimit) {
  RunDropoutTest("TrainableDropout", true, {1000}, 0.0f, false);
}

Reply:
Agree on not using the counter. Given that the whole config is just one boolean, I'll switch to use a simple vector to differentiate the fp32 and fp16 case.

---
In reply to: [437109997](https://github.com/microsoft/onnxruntime/pull/4113#discussion_r437109997) [](ancestors = 437109997)

Reply:
sure, in this case, please output which config is used when the UT failed, thx

Reply:
Updated. Thanks!

---
In reply to: [437132832](https://github.com/microsoft/onnxruntime/pull/4113#discussion_r437132832) [](ancestors = 437132832)

Reply:
nice !!

---
In reply to: [437140259](https://github.com/microsoft/onnxruntime/pull/4113#discussion_r437140259) [](ancestors = 437140259,437132832)